### ISSUE TITLE
Changing status page URL breaks the service

### ISSUE THREAD

**Issue by sundowndev** [louislam/uptime-kuma#1847]
### ⚠️ Please verify that this bug has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/uptime-kuma/security/policy)

### Description

When we set the entry page for a user to a specific status page and then we change that page's slug, it breaks the website, even when we're not logged in. I noticed this bug while the readiness probe was failing in Kubernetes. I checked that the app is listening to port 3001 and there's no migration or such stuff going on.

### 👟 Reproduction steps

1. Create a status page
2. Go to setting and set the entry page to that status page
3. Edit the status page and change the slug to something else
4. Run `curl -vvv localhost:3001` and notice the 404 http code

### 👀 Expected behavior

I'd expect it to update the user's settings as well, or just reset the entry page to default (dashboard) for any user that has this page as entry page.

### 😓 Actual Behavior

It breaks the home page, even for non-logged in users.

### 🐻 Uptime-Kuma Version

1.17.1

### 💻 Operating System and Arch

Docker louislam/uptime-kuma:1.17.1

### 🌐 Browser

Firefox

### 🐋 Docker Version

Kubernetes v1.20+

**Comment by UltraWelfare** [comment]
I've spent the last 45 minutes searching the cause of the issue.
As far as I understand this part is responsible for redirecting upon hitting the entry route `/`
https://github.com/louislam/uptime-kuma/blob/9c1ad4f8c612e924be2c2ba876a73dc24f1985d0/server/server.js#L185-L190

The first thing I've noticed that `exports` are being used to handle the information about the `entryPage` .
In the same file, later down there is a socket listener on `setSettings` which is what I believe is called when you hit the save button on the settings. Since we're in the same file it works just fine and the route code (from above) is able to pick the changes up.
https://github.com/louislam/uptime-kuma/blob/9c1ad4f8c612e924be2c2ba876a73dc24f1985d0/server/server.js#L1086-L1087

I've found that the logic for the editing the status pages resides in another folder and works correctly by doing these two things:
https://github.com/louislam/uptime-kuma/blob/9c1ad4f8c612e924be2c2ba876a73dc24f1985d0/server/socket-handlers/status-page-socket-handler.js#L229-L233

But, what is being changed here is the `server.entryPage` variable (which `server` is an `UptimeKumaServer` singleton instance). The entry route application code looks at `exports.entryPage` so it is not picked up until the next server restart.

A solution is to swap `exports.entryPage` with `UptimeKumaServer.getInstance().entryPage` so it's always correct. But this brings upon another issue that the `entryPage` of the `UptimeKumaServer` is not loaded correctly. Upon start-up it's always `dashboard`, and I couldn't find if it should be initialized somewhere...?
One solution to that that I found works is to add another line of code here that also sets the `UptimeKumaServer.getInstance().entryPage` but I'm not sure if it's the correct thing to do in terms of place in the code.
https://github.com/louislam/uptime-kuma/blob/9c1ad4f8c612e924be2c2ba876a73dc24f1985d0/server/server.js#L154-L158


By combining the change from `exports.entryPage` to `server.entryPage` and adding the one line of code that correctly instantiates the `UptimeKumaServer` entryPage, I've found that I have fixed the bug and concluded to a correct behaviour.

Let me know about it, and I can proceed with a PR.
